src/Components/Server/src/Circuits/RevalidatingServerAuthenticationStateProvider.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Security.Claims;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Components.Server
+{
+    /// <summary>
+    /// A base class for <see cref="AuthenticationStateProvider"/> services that receive an
+    /// authentication state from the host environment, and revalidate it at regular intervals.
+    /// </summary>
+    public abstract class RevalidatingServerAuthenticationStateProvider
+        : ServerAuthenticationStateProvider, IDisposable
+    {
+        private readonly ILogger _logger;
+        private readonly TimeSpan _revalidationInterval;
+        private CancellationTokenSource _loopCancellationTokenSource = new CancellationTokenSource();
+
+        /// <summary>
+        /// Constructs an instance of <see cref="RevalidatingServerAuthenticationStateProvider"/>.
+        /// </summary>
+        /// <param name="loggerFactory">A logger factory.</param>
+        /// <param name="revalidationInterval">The interval between revalidation attempts.</param>
+        public RevalidatingServerAuthenticationStateProvider(
+            ILoggerFactory loggerFactory,
+            TimeSpan revalidationInterval)
```

Comment:
It's a little unnatural seeing a timespan in a constructor parmaeter of a service. I think it would be a better design approach to just make this a property

Reply:
I started with a property, but really didn't want to supply any default value in the base class. The base can't know what sort of interval makes sense for any given scenario.

I'll try changing it to an abstract property and see how that looks.